repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/git/GitHelper.java

```diff
@@ -404,4 +406,27 @@ public void removeNotificationFromTravisYML(File directory, AbstractStep step) {
         }
     }
 
+    public void removeGhOauthFromCreatedFilesToPush(File directory, List<String> fileNames) {
+        String ghOauthPattern = "--ghOauth\\s+[\\w]+";
+        for (String fileName : fileNames) {
+            File file = new File(directory, fileName);
+
+            if (!file.exists()) {
+                getLogger().warn("The file "+file.toPath()+" does not exist.");
+            } else {
+                Charset charset = StandardCharsets.UTF_8;
+                try {
+                    String content = new String(Files.readAllBytes(file.toPath()), charset);
```

Comment:
I really don't know: do you know if it's the most efficent way of doing it? We might be manipulating logs of big size (more than 50 Mb in some cases). I just want to be sure it's the good way to do it.

Reply:
I also don't know. This was the simpler way I found that does the job, but I don't know if this is also the most efficient way. I will do some research, but I promise nothing because I don't know if I will find what we want to know.

Reply:
Based on this answer: https://stackoverflow.com/questions/326390/how-do-i-create-a-java-string-from-the-contents-of-a-file?rq=1 it looks like the good way to do it. I'll merge the PR.